src/duckling/canvas/map-editor.component.ts

```diff
@@ -218,6 +227,7 @@ export class MapEditorComponent implements AfterViewInit, OnDestroy {
         this._canvasBackgroundDisplayObject = this._buildCanvasBackground();
         this._canvasBorderDisplayObject = this._buildCanvasBorder();
         this._gridDisplayObject = this._buildGrid();
+        this._toolDisplayObject = this.tool.getDisplayObject(this.scale);
```

Comment:
You should call _redrawAllDisplayObjects when setting a new tool so that the different tool drawables will update when the tools are changed